libs/fluid-elements/virtual-scroll-container/src/lib/virtual-scroll-state.ts

```diff
@@ -0,0 +1,291 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface FluidVirtualScrollOptions {
+  itemsBatchSize?: number;
+  itemsBufferSize?: number;
+}
+
+export interface FluidVirtualScrollRenderedItemRange {
+  first: number;
+  last: number;
+}
+
+// Number of items to render each time until the container is filled up
+const VIRTUAL_SCROLL_BATCH_SIZE = 5;
+// Number of items to render in addition to the initially rendered items
+const VIRTUAL_SCROLL_BUFFER_SIZE = 3;
+
+export class VirtualScrollState {
+  /** Average item height */
+  avgItemHeight: number;
+
+  /**
+   * Index of the first and last rendered item.
+   *
+   * Important note: Never attempt to enforce to alter the object returned by
+   * this getter nor rely on its state after {@link #updateStartScrollBuffer()}
+   * or {@link #updateEndScrollBuffer()} have been called again!
+   * In case you need to store that information past this point, always create
+   * a copy right away instead.
+   * @ ChMa: pls explain again why it should be copied? I forgot what the potential issue was...
+   */
+  renderedItemsRange: FluidVirtualScrollRenderedItemRange = Object.seal({
+    first: -1,
+    last: -1,
+  });
```

Comment:
1. Why does this have to be an object? Can we just have 2 properties? `firstItemIndex` and `lastItemIndex` same as:
![image](https://user-images.githubusercontent.com/11156362/94649060-2de21880-02f4-11eb-9473-ffd9e484b898.png)

2. The whole class should be marked as internal and not be exported to the user as this is an internal state.

3. Why do we need an object seal here? It puts only overhead in the code that is not needed. Please remove
```suggestion
  renderedItemsRange: FluidVirtualScrollRenderedItemRange = {
    first: -1,
    last: -1,
  };
```